### DISCUSSION TITLE
Get some metrics / stats from tokio runtime

### DISCUSSION THREAD

**Discussion by BratSinot** [tokio-rs/tokio#5077] [Q&A] [answered]
Greetings!

Any way to get some stats from `tokio` like number of threads, task, opened connections etc?

**Comment by hds** [comment, marked as answer]
You probably want to have a look at tokio console (https://github.com/tokio-rs/console) and/or tokio-metrics (https://github.com/tokio-rs/tokio-metrics).

**Reply by BratSinot** [reply]
Thanks!

`tokio-metrics` seems is what I need (I saw it in docs, but it was in old tokio, and I think it just dropped).

**Comment by Noah-Kennedy** [comment]
@BratSinot there is an unstable API for this: https://docs.rs/tokio/latest/tokio/runtime/struct.RuntimeMetrics.html

**Reply by BratSinot** [reply]
Thanks!

I saw `stat` feature and tried to use it and missed `RuntimeMetrics`.

**Reply by Darksonn** [reply]
The `stat` feature also requires the tokio-unstable flag. See [docs.rs/tokio#unstable-features](https://docs.rs/tokio/latest/tokio/#unstable-features) for details on how to enable unstable features.

**Reply by Noah-Kennedy** [reply]
@BratSinot the parts of `tokio-metrics` that you would want are built on top of `RuntimeMetrics`, and thus require the unstable flag as well.
